distributed/scheduler.py

```diff
@@ -422,6 +426,17 @@ def start(self, addr_or_port=8786, start_queues=True):
             for k, v in self.services.items():
                 logger.info("%11s at: %25s", k, '%s:%d' % (self.ip, v.port))
 
+        if self.scheduler_file:
+            with ignoring(ImportError):
+                with open(self.scheduler_file, 'w') as f:
+                    json.dump(self.identity(), f, indent=2)
+
+            fn = self.scheduler_file  # remove file when we close the process
+            def del_scheduler_file():
+                if os.path.exists(fn):
+                    os.remove(fn)
+            atexit.register(del_scheduler_file)
```

Comment:
We should really backport `wearef.finalize` from Python 3. I can do that in a separate PR.